h2o-algos/src/main/java/hex/tree/isofor/IsolationForestModel.java

```diff
@@ -78,8 +90,13 @@ public ModelCategory getModelCategory() {
    *  subclass scoring logic. */
   @Override protected double[] score0(double[] data, double[] preds, double offset, int ntrees) {
     super.score0(data, preds, offset, ntrees);
-    if (ntrees >= 1) preds[1] = preds[0] / ntrees;
-    preds[0] = normalizePathLength(preds[0]);
+    boolean outputAnomalyFlag = outputAnomalyFlag();
+    int off = outputAnomalyFlag ? 1 : 0;
+    if (ntrees >= 1) 
+      preds[off + 1] = preds[0] / ntrees;
+    preds[off] = normalizePathLength(preds[0]);
+    if (outputAnomalyFlag)
+      preds[0] = preds[1] >= _output._defaultThreshold ? 1 : 0; 
```

Comment:
```suggestion
    final int off;
    if (outputAnomalyFlag()) {
      off = 1;
    } else {
      off = 0;
    }
    if (ntrees >= 1) {
      preds[off + 1] = preds[0] / ntrees;
      preds[0] = preds[1] >= _output._defaultThreshold ? 1 : 0;
    }
    preds[off] = normalizePathLength(preds[0]);
```

Nit: Checks the `outputAnomalyFlag` only once while not creating extra local variable. Also avoids the `if (ntrees >= 1)` without curly braces.

Reply:
thank you for the suggestion, the proposed code is incorrect - you are overwriting preds[0] and then using it instead of the original value

Reply:
Oh I see, thanks. Overlooked that one. My apologies.